src/hooks/useLogin.tsx

```diff
@@ -0,0 +1,28 @@
+import KyberOauth2 from '@kybernetwork/oauth2'
+import { useEffect, useRef } from 'react'
+
+import { OAUTH_CLIENT_ID } from 'constants/env'
+
+const useLogin = () => {
+  // prevent spam because react 18 strict mode
+  const requestingAnonymous = useRef(false)
```

Comment:
I didnt get your idea. why isn't it a loading state?

Reply:
Isn't empty array dependencies solved this? Why do you need an additional ref

Reply:
ah. nice, move it into index.tsx.